VideoApi/VideoApi.DAL/Queries/GetHostsInHearingsTodayQuery.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using VideoApi.DAL.Queries.Core;
+using VideoApi.Domain;
+using VideoApi.Domain.Enums;
+
+namespace VideoApi.DAL.Queries
+{
+    public class GetHostsInHearingsTodayQuery : IQuery
+    {
+        public bool JudgesOnly { get; set; }
+
+        public GetHostsInHearingsTodayQuery(bool judgesOnly = false)
+        {
+            JudgesOnly = judgesOnly;
+        }
+    }
+
+    public class GetHostsInHearingsTodayQueryHandler : IQueryHandler<GetHostsInHearingsTodayQuery, List<Conference>>
+    {
+        private readonly VideoApiDbContext _context;
+
+        public GetHostsInHearingsTodayQueryHandler(VideoApiDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<Conference>> Handle(GetHostsInHearingsTodayQuery query)
+        {
+            var today = DateTime.Today;
+            var tomorrow = DateTime.Today.AddDays(1);
+
+            var queryToExecute = _context.Conferences
+                .Include(x => x.Participants)
+                .AsNoTracking()
+                .Where(x => x.ScheduledDateTime >= today && x.ScheduledDateTime < tomorrow);
+
+            if (query.JudgesOnly)
+                queryToExecute = queryToExecute.Where(x => x.Participants.Any(p => p.UserRole == UserRole.Judge
+                 && (p.State == ParticipantState.InHearing || p.State == ParticipantState.Available)));
+            else
+                queryToExecute = queryToExecute
+                    .Where(x => x.Participants.Any(p => (p.UserRole == UserRole.Judge || p.UserRole == UserRole.StaffMember)
+                 && (p.State == ParticipantState.InHearing || p.State == ParticipantState.Available)));
```

Comment:
Any way the duplication here could be reduced? Perhaps a list of user roles that are valid that only contains judge if true and judge and staff memeber if false

Reply:
Nice. Thanks. My brain wasn't on form yesterday and I was struggling to come up with a cleaner way. This is perfect